crates/testing/src/overall_safety_task.rs

```diff
@@ -159,9 +159,7 @@ impl<TYPES: NodeType, I: TestableNodeImplementation<TYPES>> TestTaskState
                 block_size: maybe_block_size,
             } => {
                 // Skip the genesis leaf.
-                if leaf_chain.len() == 1
-                    && leaf_chain[0].0.get_view_number() == TYPES::Time::genesis()
-                {
+                if leaf_chain.last().unwrap().0.get_view_number() == TYPES::Time::genesis() {
```

Comment:
I'm wondering whether we should still check the length in case somehow the leaf chain contains a non-genesis leaf at the second index.

Reply:
Oops just saw that you changed `[0]` to `last()`. Makes sense now!